### PR TITLE
Add "zinter" script

### PR DESCRIPTION
**Pull request by yohanboniface**
Taking advantage of the new script possibility :)

Redis has sinter and sinterstore, but only zinterstore.
We create our own zinter for cleaner API and a bit of speedup.

@vinyll @davidbgk r?

### CODE REVIEW

**Comment by vinyll** [comment]
_zinter.lua_ o.0

File: tests/test_scripts.py [hunk 1]
```diff
@@ -10,3 +10,14 @@ def test_manual_scan(factory):
                                   args=[2])
     assert results == ['d|{}'.format(street1['id']).encode(),
                        'd|{}'.format(street2['id']).encode()]
+
+
+def test_zinter(factory):
+    factory(name="rue de la monnaie", city="Vitry")
+    factory(name="La monnaye", city="Saint-Loup-Cammas")
+    street1 = factory(name="rue de la monnaie", city="Paris", importance=1)
+    street2 = factory(name="rue de la monnaie", city="Condom", importance=0.9)
```
Comment (vinyll) [inline tests/test_scripts.py]: I'd rather write

```python 
streets = (
    factory(name="rue de la monnaie", city="Paris", importance=1),
    factory(name="rue de la monnaie", city="Condom", importance=0.9)
)
…
streets[0]['id']
…
```

File: addok/core.py [hunk 2]
```diff
@@ -205,9 +205,7 @@ def intersect(self, keys, limit=0):
             if len(keys) == 1:
                 ids = DB.zrevrange(keys[0], 0, limit - 1)
             else:
-                DB.zinterstore(self.pid, set(keys))
-                ids = DB.zrevrange(self.pid, 0, limit - 1)
-                DB.delete(self.pid)
+                ids = scripts.zinter(keys=set(keys), args=[self.pid, limit-1])
```
Comment (vinyll) [inline addok/core.py]: Personally I'm not familiar with Redis call scripts. Not sure how to review this part.

Comment (davidbgk) [inline addok/core.py]: @vinyll you missed the previous PR ;) https://github.com/addok/addok/pull/199

File: tests/test_scripts.py [hunk 3]
```diff
@@ -10,3 +10,14 @@ def test_manual_scan(factory):
...
+    results = scripts.manual_scan(keys=['w|monnaie', 'w|rue', 'w|de'],
+                                  args=[2])
+    assert results == ['d|{}'.format(street1['id']).encode(),
+                       'd|{}'.format(street2['id']).encode()]
```
Comment (davidbgk) [inline tests/test_scripts.py]: I would test that passing `args=[3]` returns a street without the `importance` set.

File: tests/test_scripts.py (hunk 1 above)
Comment (yohanboniface) [inline tests/test_scripts.py]: Changed :)

**Pull request merged**